docs/Course Wiki/ECE Course Offerings/ECE120.md

```diff
@@ -0,0 +1,55 @@
+
+## Course Tips
+
+Some of the topics covered in this course may be hard to grasp immediately, even after attending lectures and discussions, so it is highly recommended to start the homework early and attend TA and CA office hours with questions about the homeworks and any topics you may be confused on. The office hours right before the homework is due are often crowded - instead, attempt the homework early and go to the office hours right after homework is due.
+
+Do not be afraid to ask questions in lecture and discussion! ECE120 covers a vast amount of topics to provide a broad overview of computing, so understanding the material will be helpful for every future computing-related ECE course in some way or another. Make it a goal to ask a question every lecture (this is easily doable if you pay attention, except maybe the first lecture).
+
+## Life After
+
+After completing ECE120, ECE students are encouraged to take ECE220, the second core computer engineering course, as soon as possible. ECE220 builds onto concepts introduced at the tail end of ECE120, which in turn build on concepts covered throughout the rest of the course; therefore, having a strong understanding of ECE120 material is important prior to taking ECE220. In due time after taking the core ECE courses, ECE385 offers a continuation of logical design elements, as well as hardware and software implementations of applicable scenarios. ECE411 offers an even more in-depth continuation of the computer organization concepts covered in ECE120 and ECE385, and is generally taken by students in their senior year after completing the rest of the computer engineering core classes.
+
+## Infamous Topics
+
+- Sequential logic design: This is covered about halfway through the course, and alongside combinational design forms the backbone of every computing device imaginable. The idea of sequential design centers around feeding combinational gate outputs back into each other and observing changes over time with a varying clock. This can be a bit obtuse to understand, but is vital to know for designing finite state machines, memories, and eventually a full CPU, which the class builds up to in the coming weeks.
+- Finite state machines: Covered soon after sequential logic design, finite state machines (FSMs) are the core of a computing device as they dictate what decision to make given a current state of operation and inputs. Finite state machine design is often diffucult for students to understand and apply to given scenarios, made even harder if they don't properly understand sequential logic, which is what FSMs build upon. This is why, for good reason, many students consider finite state machines to be one of the hardest, if not the hardest part of the course.
```

Comment:
We may wanna add serialization and IEEE-754 Floating Point bc I feel like students struggle with that